### PR TITLE
ON-807 # Improved error handling for API calls

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: docs/forms.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+# OneBlink SDK | Forms
+
+## Constructor
+
+| Parameter | Required | Type | Description
+|---|---|---|---|
+| `options.accessKey` | Yes | `string` | Access key provided by OneBlink. |
+| `options.secretKey` | Yes | `string` | Secret key provided by OneBlink. |
+| `options.formsRendererOrigin` | No | `string` | The Origin of the Forms Renderer. Should only be used for development and testing purposes. |
```
Comment (kizaonline) [inline docs/forms.md]: Should we even expose these options to them or leave them undocumented?

File: docs/forms.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+## Functions
+
+### `generateFormUrl(formId, externalId)`
+
+### Parameters
+
+| Parameter | Required | Type | Description
+|---|---|---|---|
+| `formId` | Yes | `number` | The exact id of the form you wish to generate a URL for |
+| `externalId` | No | `string` | The external identifier of the form or form submission you wish to use, this will be given back to you in the query string |
```
Comment (kizaonline) [inline docs/forms.md]: of the form or form submission? Should it say something more along lines of this id will be returned to you with the submissionId for later retrieving the submission data?

File: docs/forms.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+```
+
+### `getSubmissionData(formId, submissionId)`
+
+### Parameters
+
+| Parameter | Required | Type | Description
+|---|---|---|---|
+| `formId` | Yes | `number` | The exact id of the form you wish to generate a URL for |
+| `submissionId` | Yes | `string` | The submission id generated after a successful form submission |
```
Comment (kizaonline) [inline docs/forms.md]: Mention that it is returned via the callback?

File: docs/forms.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  })
+```
+
+### `search(organisationId, isPublished)`
+
+### Parameters
+
+| Parameter | Required | Type | Description
+|---|---|---|---|
+| `organisationId` | Yes | `string` | The exact id of the organisation you wish to return a list of forms from |
```
Comment (kizaonline) [inline docs/forms.md]: Do we need to explain how to get this value?

File: README.md [hunk 5]
```diff
@@ -10,84 +10,4 @@ npm install @oneblink/sdk --save
...
-      "description": "a form",
-      "organisationId": "0101010101010",
-      "elements": [],
-      "isAuthenticated": false,
-      "isPublished": true,
-      "submissionEvents": []
-    }
-  ]
-```
+-   [Documentation](./docs/README.md)
```
Comment (kizaonline) [inline README.md]: Can we include some docs about the overall process/flow?

File: docs/forms.md (hunk 1 above)
Comment (mymattcarroll) [reply docs/forms.md]: im easy, what do you think?

File: docs/forms.md (hunk 2 above)
Comment (mymattcarroll) [reply docs/forms.md]: 👍

File: docs/forms.md (hunk 1 above)
Comment (kizaonline) [reply docs/forms.md]: Undocumented

File: README.md (hunk 5 above)
Comment (mymattcarroll) [reply README.md]: Agreed, lets pair and do this.

File: docs/forms.md (hunk 4 above)
Comment (mymattcarroll) [reply docs/forms.md]: Lets pair up and decide

**Pull request merged**